likeminds-feed-reactnative-integration/components/LMMedia/LMImage/index.tsx

```diff
@@ -23,44 +24,63 @@ const LMImage = React.memo(
   }: LMImageProps) => {
     const [loading, setLoading] = useState(true);
     const [error, setError] = useState(false);
-    const [heightCalculated, setHeightCalculated] = useState(0);
+    const [heightCalculated, setHeightCalculated] = useState(height ?? 400);
     const [desiredAspectRatio, setDesiredAspectRatio] = useState(0);
 
-    useEffect(() => {
-      Image.getSize(
-        imageUrl,
-        (width, height) => {
-          const ScreenWidth = Dimensions.get("window").width;
-          const desiredAspectRatio = width > height ? 1.91 : 0.8;
-          const heightCalculated = ScreenWidth * (1 / desiredAspectRatio);
-          setHeightCalculated(heightCalculated);
-          setDesiredAspectRatio(desiredAspectRatio);
-        },
-        (error) => {
-          console.error("Error getting image size:", error);
-        }
-      );
+    useLayoutEffect(() => {
+      if (!height || !width) {
+
+        Image.getSize(
+          imageUrl,
+          (width, height) => {
+            const ScreenWidth = Dimensions.get("window").width;
+            const desiredAspectRatio = width > height ? 1.91 : 0.8;
+            const heightCalculated = ScreenWidth * (1 / desiredAspectRatio);
+
+            setHeightCalculated(heightCalculated);
+            setDesiredAspectRatio(desiredAspectRatio);
+          },
+          (error) => {
+            console.error('Failed to get image size:', error);
+          }
+        );
+
+        return;
+
+      } else {
+
+        const ScreenWidth = Dimensions.get("window").width;
```

Comment:
```suggestion
      const screenWidth = Dimensions.get("window").width;
```
1. Rename variable
2. Why are we calculating screenWidth again and again, calculate once and store it for better performance. Change this approach.